source/Background/Controller.js

```diff
@@ -149,9 +149,28 @@ backgroundController.exposeController('disconnect', async (opts, url) => secureC
     const apps = response?.[keyring.currentWalletId]?.apps;
 
     if (apps?.[url]) {
-      const newApps = removeAppByURL({ apps, url });
+      //const newApps = removeAppByURL({ apps, url });
 
+      const date = new Date().toISOString();
+
+      const newApps = {
+        ...apps,
+        [url]: {
+          ...apps[url],
+          status: CONNECTION_STATUS.disconnected,
+          date,
+          whitelist,
+          events: [
+            ...apps[url].events,
+            {
+              status: CONNECTION_STATUS.disconnected,
+              date,
+            }
+          ]
+        },
+      };
       storage.set({ [keyring.currentWalletId]: { apps: newApps } });
+
```

Comment:
I'd remove the comment and make all of this a func like `addDisconnectEntry` or sth like that just to keep it similar to how it was.